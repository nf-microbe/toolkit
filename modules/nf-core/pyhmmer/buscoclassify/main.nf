process PYHMMER_BUSCOCLASSIFY {
    tag "${meta.id}"
    label 'process_high'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'oras://community.wave.seqera.io/library/pyhmmer_pandas:9f16bf144bcd5b1a':
        'community.wave.seqera.io/library/pyhmmer_pandas:842847931f77e2ab' }"

    input:
    tuple val(meta), path(faa)
    path archaea_hmms
    path archaea_cutoffs
    path bacteria_hmms
    path bacteria_cutoffs

    output:
    tuple val(meta), path("${prefix}.busco_hmms.tsv")       , emit: hits
    tuple val(meta), path("${prefix}.busco_markers.tsv")    , emit: markers
    path "versions.yml"                                     , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    pyhmmer_buscoclassify.py \\
        --input ${faa} \\
        --archaea_hmms ${archaea_hmms} \\
        --archaea_cutoffs ${archaea_cutoffs} \\
        --bacteria_hmms ${bacteria_hmms} \\
        --bacteria_cutoffs ${bacteria_cutoffs} \\
        --prefix ${prefix} \\
        ${args}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$( python --version | sed 's/Python //' )
        pandas: \$(python -c "import pandas; print(pandas.__version__)")
        pyhmmer: \$(python -c "import pyhmmer; print(pyhmmer.__version__)")
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}.busco_hmms.tsv
    touch ${prefix}.busco_markers.tsv

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$( python --version | sed 's/Python //' )
        pandas: \$(python -c "import pandas; print(pandas.__version__)")
        pyhmmer: \$(python -c "import pyhmmer; print(pyhmmer.__version__)")
    END_VERSIONS
    """
}
