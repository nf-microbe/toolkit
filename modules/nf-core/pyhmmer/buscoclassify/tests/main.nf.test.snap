{
    "bacteroides_fragilis - faa - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.busco_hmms.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.busco_markers.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    "versions.yml:md5,c221730f75d0b62da616d2951b2d3082"
                ],
                "hits": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.busco_hmms.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "markers": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.busco_markers.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,c221730f75d0b62da616d2951b2d3082"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-12T15:13:35.684961508"
    },
    "bacteroides_fragilis - faa": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.busco_hmms.tsv:md5,fd9af19eb717574f3638232394bd6d3b"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.busco_markers.tsv:md5,9eaebf4012abdb6233e98d471f9b9578"
                    ]
                ],
                "2": [
                    "versions.yml:md5,c221730f75d0b62da616d2951b2d3082"
                ],
                "hits": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.busco_hmms.tsv:md5,fd9af19eb717574f3638232394bd6d3b"
                    ]
                ],
                "markers": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.busco_markers.tsv:md5,9eaebf4012abdb6233e98d471f9b9578"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,c221730f75d0b62da616d2951b2d3082"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-12T15:15:59.550326459"
    }
}